Import-VIRole.ps1

```diff
@@ -0,0 +1,114 @@
+        [Parameter(Mandatory = $true,Position = 1,HelpMessage = 'Path to the JSON file containing permissions')]
+        [ValidateNotNullorEmpty()]
+        [String]$Permission,
+        [Parameter(Mandatory = $true,Position = 2,HelpMessage = 'vCenter Server IP or FQDN')]
+        [ValidateNotNullorEmpty()]
+        [String]$vCenter
+    )
+
+    Process {
+
+        Write-Verbose -Message 'Importing required modules and snapins'
+        $powercli = Get-PSSnapin -Name VMware.VimAutomation.Core -Registered
+        try 
+        {
+            switch ($powercli.Version.Major) {
+                {
+                    $_ -ge 6
+                }
+                {
+                    Import-Module -Name VMware.VimAutomation.Core -ErrorAction Stop
+                    Write-Verbose -Message 'PowerCLI 6+ module imported'
+                }
+                5
+                {
+                    Add-PSSnapin -Name VMware.VimAutomation.Core -ErrorAction Stop
+                    Write-Warning -Message 'PowerCLI 5 snapin added; recommend upgrading your PowerCLI version'
+                }
+                default 
+                {
+                    throw 'This script requires PowerCLI version 5 or later'
+                }
+            }
+        }
+        catch 
+        {
+            throw $_
+        }
+
+        
+        Write-Verbose -Message 'Allowing untrusted SSL certs'
+        Add-Type -TypeDefinition @"
+	    using System.Net;
+	    using System.Security.Cryptography.X509Certificates;
+	    public class TrustAllCertsPolicy : ICertificatePolicy {
+	        public bool CheckValidationResult(
+	            ServicePoint srvPoint, X509Certificate certificate,
+	            WebRequest request, int certificateProblem) {
+	            return true;
+	        }
+	    }
+"@
+        [System.Net.ServicePointManager]::CertificatePolicy = New-Object -TypeName TrustAllCertsPolicy
+
+        Write-Verbose -Message 'Ignoring self-signed SSL certificates for vCenter Server (optional)'
+        $null = Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -DisplayDeprecationWarnings:$false -Scope User -Confirm:$false
+
+        Write-Verbose -Message 'Connecting to vCenter'
+        try 
+        {
+            $null = Connect-VIServer -Server $vCenter -ErrorAction Stop -Session ($global:DefaultVIServers | Where-Object -FilterScript {
+                    $_.name -eq $vCenter
+            }).sessionId
+        }
+        catch 
+        {
+            throw 'Could not connect to vCenter'
+        }
+
+        Write-Verbose -Message 'Check to see if the role exists'
+        if (Get-VIRole -Name $Name -ErrorAction SilentlyContinue) 
+        {
+            throw 'Role already exists'
+        }
+
+        Write-Verbose -Message 'Read the JSON file'
+        try
+        {
+            $null = Test-Path $Permission
+            [array]$PermArray = Get-Content -Path $Permission -Raw | ConvertFrom-Json
+        }
+        catch
```

Comment:
I've had some issues in the past where errors don't result in halting (hence the catch/throw) and dislike having to explicitly state -ErrorAction:$blowup (or whatever it is). But in this case, no, probably not required.